Content.Shared/Clothing/LoadoutSystem.cs

```diff
@@ -120,19 +120,30 @@ private void OnMapInit(EntityUid uid, LoadoutComponent component, MapInitEvent a
         if (component.StartingGear != null)
         {
             var gear = _protoMan.Index(_random.Pick(component.StartingGear));
-            _station.EquipStartingGear(uid, gear);
-            return;
+            _station.EquipStartingGear(uid, gear, false);
         }
 
         if (component.RoleLoadout == null)
             return;
 
-        // ...otherwise equip from role loadout
-        var id = _random.Pick(component.RoleLoadout);
-        var proto = _protoMan.Index(id);
-        var loadout = new RoleLoadout(id);
-        loadout.SetDefault(GetProfile(uid), _actors.GetSession(uid), _protoMan, true);
-        _station.EquipRoleLoadout(uid, loadout, proto);
+        // Equip role loadout(s). The component's AddAllRoleLoadouts datafield determines
+        // if all loadouts should be applied, or only a single random one.
+        var idList = new List<ProtoId<RoleLoadoutPrototype>>();
+        if (!component.AddAllRoleLoadouts)
+            idList.Add(_random.Pick(component.RoleLoadout));
```

Comment:
Won't this mean if you try to give a ghost role a random one from the list of loadouts that the species tank loadouts wont work?

Reply:
It does. But I think the random way makes no sense for Loadouts anyway so I don't concern myself with the applications

Reply:
That's not ideal as they would be really nice for things like visiting shuttles or other somewhat fluff roles.

Reply:
Describe to me how you picture this working/being used, because I don't get it. What Loadouts do you want to randomly pick? Can't you just put that in the StartingGear instead, which WILL be a random roll between the ones you select regardless of which way you are using the RoleLoadout list?

Reply:
> Can't you just put that in the StartingGear instead, which WILL be a random roll between the ones you specified.


Ah, I thought this was what you had changed. That said though, I now don't know why you would need to ever set AddAllRoleLoadouts = true.

Reply:
I don't know either, but since that was the original implementation, apparently very intentionally, I left the option in

Reply:
> Won't this mean if you try to give a ghost role a random one from the list of loadouts that the species tank loadouts wont work?

Actually, I was wrong. It picks a random RoleLoadout, but each of those can include multiple LoadoutGroups. The survival stuff is in fact in a LoadoutGroup, so it's fine, it can be added. In fact, I am probably going to yank out the entire "include all RoleLoadouts" functionality because I have come to see that it's rather superfluous